internal/services/apimanagement/api_management_api_tag_description_resource.go

```diff
@@ -0,0 +1,166 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2021-08-01/apimanagement"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceApiManagementApiTagDescription() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Read:   resourceApiManagementApiTagDescriptionRead,
+		Update: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Delete: resourceApiManagementApiTagDescriptionDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ApiTagDescriptionsID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"tag_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.TagID,
+			},
+
+			"api_name": schemaz.SchemaApiManagementApiName(),
```

Comment:
and then we don't need this
```suggestion

```

Reply:
`tag_id` contains is `resource_group`, `apim_name`  but does not have the API name, we need to specific the API name which will be used to associate with the tag